src/middleware.ts

```diff
@@ -0,0 +1,54 @@
+import { NextRequest, NextResponse } from "next/server";
+import {
+  ERROR_INVALID_SESSION_SCHEMA,
+  SESSION_COOKIE_NAME,
+  verifySession,
+} from "./lib/tools/session";
+
+export async function middleware(request: NextRequest) {
+  const sessionCookie = request.cookies.get(SESSION_COOKIE_NAME)?.value;
+
+  if (sessionCookie) {
+    try {
+      await verifySession(sessionCookie);
+    } catch (err) {
+      if (
+        !(err instanceof Error) ||
+        err.message !== ERROR_INVALID_SESSION_SCHEMA
+      ) {
+        console.error(err);
+        return NextResponse.next();
+      }
+
+      if (request.nextUrl.pathname !== "/signin" && request.method === "GET") {
```

Comment:
Not sure if we need this right now do we? Also, wouldn't it redirect the landing and signup flows?

Reply:
What if we are on `/signin`? That'd cause an infinite loop of redirects. I guess I can add the `/` and `/signup` as well to the check.

Reply:
Sorry, I meant the whole redirect logic isn't needed right now, not just the conditional.

Reply:
I see. So, I should just delete the malformed cookie and not redirect.